apps/api/src/app/inbox/usecases/delete-subscription/delete-subscription.usecase.ts

```diff
@@ -1,23 +1,37 @@
 import { Injectable, NotFoundException } from '@nestjs/common';
 import { InstrumentUsecase } from '@novu/application-generic';
-import { PreferencesRepository, TopicSubscribersRepository } from '@novu/dal';
+import { PreferencesRepository, TopicRepository, TopicSubscribersRepository } from '@novu/dal';
 import { PreferencesTypeEnum } from '@novu/shared';
 import { DeleteTopicSubscriptionCommand } from './delete-subscription.command';
 
 @Injectable()
 export class DeleteTopicSubscription {
   constructor(
     private topicSubscribersRepository: TopicSubscribersRepository,
-    private preferencesRepository: PreferencesRepository
+    private preferencesRepository: PreferencesRepository,
+    private topicRepository: TopicRepository
   ) {}
 
   @InstrumentUsecase()
   async execute(command: DeleteTopicSubscriptionCommand): Promise<{ success: boolean }> {
+    const topic = await this.topicRepository.findTopicByKey(
+      command.topicKey,
+      command.organizationId,
+      command.environmentId
+    );
+
+    if (!topic) {
+      throw new NotFoundException(`Topic with key ${command.topicKey} not found`);
+    }
```

Comment:
aligned the delete endpoint with the rest and updated the usecase to perform topic check